### PR TITLE
[feature] WPN-97 gestion des plugins

### PR DESCRIPTION
**Pull request by rosamaggi**
https://erpdev.atlassian.net/browse/WPN-97

- Delete useless variables form CRD and add some new ones
- No more secret hard coded: https://github.com/epfl-si/wp-operator/pull/14
- Add `add_header Cache-Control "129600, public";` in the nginx configuration for the cache of images
- WPN-116 Wait until the database and user completes to run ensure-wordpress-and-theme.php: https://erpdev.atlassian.net/browse/WPN-116
- restore_wordpress_from_os3
- Add all plugins installation and configuration with options
- Rename wpn_kopf.py --> wp_operator.py https://erpdev.atlassian.net/browse/WPN-211

### CODE REVIEW

File: wp-operator.py [hunk 1]
```diff
@@ -356,11 +385,17 @@ def create_ingress(self, path):
 location ~ (wp-includes|wp-admin|wp-content/(plugins|mu-plugins|themes))/ {{
     rewrite .*/((wp-includes|wp-admin|wp-content/(plugins|mu-plugins|themes))/.*) /$1 break;
     root /wp/6/;
+    location ~* \.(ico|pdf|flv|jpg|jpeg|png|gif|swf|svg)$ {{
```
Comment (ponsfrilus) [inline wp-operator.py:388]: I know this was done this way before, but I vote to remove `flv` and `swf` (lol) and add some more modern files such as `apng`, `avif` and `webp`.

File: wp-operator.py [hunk 2]
```diff
@@ -561,6 +629,29 @@ def delete_site(self, spec):
       self.delete_custom_object_mariadb(self.prefix['grant'], "grants")
       self.delete_ingress()
 
+  # def update_fn(self, spec, logger):
```
Comment (ponsfrilus) [inline wp-operator.py:632]: Dead code?

File: wp-operator.py [hunk 3]
```diff
@@ -594,7 +685,7 @@ def ensure_wp_crd_exists(cls):
           logging.error(f"Error verifying CRD file: {e}")
       return False
 
-    
+
```
Comment (ponsfrilus) [inline wp-operator.py:688]: Drop this line?

**Review by ponsfrilus — COMMENTED** [review]
Please see comments. Note that this is way to big of a PR and it would be more efficient to split things in smaller PR!

File: ensure-wordpress-and-theme.php [hunk 4]
```diff
@@ -178,23 +196,85 @@ function ensure_theme ( $options ) {
...
+ensure_admin_user("admin", "admin@exemple.com", generate_random_password());
 ensure_site_title( $options );
 ensure_tagline( $options );
 ensure_theme( $options );
-ensure_plugins( $options );
+delete_default_pages_and_posts();
+ensure_plugins();
 
-echo "WordPress successfully installed";
+echo "WordPress and plugins successfully installed";
```
[the remark this answers is not in the document]
Comment (ponsfrilus) [reply ensure-wordpress-and-theme.php:280]: Saw the the corresponding change is done in the operator.

File: plugins/Plugin.php [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+		}
+	}
+
+	public function getPluginPath(): string
+	{
+		return $this->pluginPath;
+	}
+
+	public function getSecretValue($secretName): string {
+		$myfile = fopen("$this->secrets_dir/$secretName", "r") or die("Unable to open file!");
```
[the remark this answers is not in the document]
Comment (rosamaggi) [reply plugins/Plugin.php]: Yes, see my other comment

File: ensure-wordpress-and-theme.php [hunk 6]
```diff
@@ -178,23 +196,85 @@ function ensure_theme ( $options ) {
...
+  for ($i = 0; $i < 32; ++$i) {
+    $str .= $keyspace[random_int(0, $max)];
+  }
+  return $str;
 }
 
 ensure_db_schema();
 ensure_other_basic_wordpress_things( $options );
-ensure_admin_user("admin", "admin@exemple.com", "secret");
+ensure_admin_user("admin", "admin@exemple.com", generate_random_password());
```
[the remark this answers is not in the document]
Comment (Azecko) [reply ensure-wordpress-and-theme.php]: Resolved.

**Review by ponsfrilus — APPROVED** [review]
:+1:

**Pull request merged**